plugins/out_opentelemetry/opentelemetry.c

```diff
@@ -107,7 +122,9 @@ static int http_post(struct opentelemetry_context *ctx,
                             key->str, flb_sds_len(key->str),
                             val->str, flb_sds_len(val->str));
     }
-
+    if (compressed == FLB_TRUE) {
+        flb_http_set_content_encoding_gzip(c);
+    }
     ret = flb_http_do(c, &b_sent);
     if (ret == 0) {
```

Comment:
Your valgrind report says there is a memory leak:

```
==407985== LEAK SUMMARY:
==407985==    definitely lost: 925,638 bytes in 21 blocks
```

looking at the changes it seems like the new buffer allocated for the compressed gzip is not being free, so on every flush with gzip, you are leaking the allocated memory.

Reply:
ok thanks for the help. It should be fixed.